### DISCUSSION TITLE
Issues with `remote_user_guard` when using Traefik forwardAuth with Authentik

### DISCUSSION THREAD

**Discussion by black-dragon74** [firefly-iii/firefly-iii#9283] [Firefly III] [answered]
Hi, I am using version 6.1.21 of Firefly III with `remote_user_guard` enabled. The application sits behind Traefik reverse proxy which has forwardAuth setup for the host using Authentik.

I am trying to access the API with a PAT.

I have excluded the api routes with regex: `^/api` and can confirm that Traefik does not redirect for authentication.

I am able to login and do everything on the website but if I try accessing the api routes using my PAT as Bearer token, I get `{"message":"Unauthenticated","exception":"AuthenticationException"}` 

Here is my middleware for traefik:

```yaml
http:
  middlewares:
    authentik:
      forwardAuth:
        address: http://authentik-server:9000/outpost.goauthentik.io/auth/traefik
        trustForwardHeader: true
        authResponseHeaders:
          - X-authentik-username
          - X-authentik-groups
          - X-authentik-email
          - X-authentik-name
          - X-authentik-uid
          - X-authentik-jwt
          - X-authentik-meta-jwks
          - X-authentik-meta-outpost
          - X-authentik-meta-provider
          - X-authentik-meta-app
          - X-authentik-meta-version
          - Authorization
```

Related Firefly config

```env
AUTHENTICATION_GUARD=remote_user_guard
AUTHENTICATION_GUARD_HEADER=HTTP_X_AUTHENTIK_EMAIL
AUTHENTICATION_GUARD_EMAIL=HTTP_X_AUTHENTIK_EMAIL
APP_URL=https://<fqdn>
```

If I remove the forwardAuth middleware, the API starts working as expected.

I am trying to debug this for past 2 hours or so but to no avail. I would appreciate any pointers.

Regards

**Comment by black-dragon74** [comment, marked as answer]
Alright, I got hold of the bug. As you can see in my middlewares I have included `Authorization` inside `authResponseHeaders`.

And in our case, since we are skipping authentik for api routes, traefik will remove the `Authorization` header that we provide as all headers listed under `authResponseHeaders` are expected to come from our auth service (Authentik)

This leads to the header not being sent to Firefly at all and hence the error saying `Unauthenticated`.

TLDR; remove `Authorization` from `authResponseHeaders` and it will work just fine.

This leads to another problem though, I use some services that use HTTP_BASIC_AUTH and that requires the `Authorization` header. I would love to hear your thoughts on this (unrelated) issue :(

Regards

**Reply by JC5** [reply]
> TLDR; remove Authorization from authResponseHeaders and it will work just fine.

Is this not a setting you can do per service, rather than global?
